process/smartContract/scQueryService.go

```diff
@@ -88,6 +95,19 @@ func (service *SCQueryService) executeScCall(query *process.SCQuery, gasPrice ui
 	log.Trace("executeScCall", "function", query.FuncName, "numQueries", service.numQueries)
 	service.numQueries++
 
+	shouldEarlyExitBecauseOfSyncState := query.ShouldBeSynced && service.bootstrapper.GetNodeState() == common.NsNotSynchronized
+	if shouldEarlyExitBecauseOfSyncState {
+		return nil, process.ErrNodeIsNotSynced
+	}
+
+	shouldCheckRootHashChanges := query.SameScState
+	rootHashBeforeExecution := make([]byte, 0)
+	var err error
```

Comment:
This declaration seems unnecessary, is it needed?

Reply:
removed. was used before, but forgot to remove after refactoring